packages/bueno/project.json

```diff
@@ -1 +1,109 @@
-{}
+{
+  "namedInputs": {
+    "tests": ["{projectRoot}/**/*.test.ts"],
+    "source": [
+      "{projectRoot}/esbuild.mjs",
+      "{workspaceRoot}/tsconfig.json",
+      "{projectRoot}/tsconfig.json",
+      "{projectRoot}/src/**/*([^.]).!(test.ts)"
+    ]
+  },
+  "targets": {
+    "clean": {
+      "dependsOn": [],
+      "inputs": [],
+      "outputs": [],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "rimraf dist",
+        "cwd": "packages/bueno"
+      }
+    },
+    "cached:build:esbuild": {
+      "dependsOn": ["^build"],
+      "inputs": ["^source", "source"],
+      "outputs": ["{projectRoot}/dist"],
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "node esbuild.mjs",
+        "cwd": "packages/bueno"
+      }
+    },
+    "cached:build:definitions": {
+      "dependsOn": ["cached:build:esbuild"],
```

Comment:
Technically, `tsc -d --emitDeclarationOnly --declarationDir dist/definitions` is able to run successfully without running `esbuild` first.

However, this task emits its declarations inside the `dist/definitions/` folder and `esbuild` emits its build artifacts in the `dist/` folder. If we run both tasks in parallel or in a varying order, the `esbuild` task's cache will randomly contain or not contain the definitions, and may occasionally delete the `dist/definitions/` directory (which did happen while I was testing, until I added this dependency).

If we really want those tasks to run in parallel, I could emit the definitions in a separate folder and then add another task which copies them into the `dist` folder.

**Question**: do you think I should do it or not?